include/bitcoin/bitcoin/chain/header.hpp

```diff
@@ -151,7 +151,7 @@ class BC_API header
     uint32_t nonce() const;
     void set_nonce(uint32_t value);
 
-    hash_digest hash() const;
+    hash_digest hash(bool litecoin=false) const;
```

Comment:
This is getting ugly.

Reply:
As I wrote in the commit msg, I was trying to stay consistent with how it's done for the `retarget` parameter where [similar ugliness](https://github.com/libbitcoin/libbitcoin/blob/81b61747ec29a45ba43f08c437c5fa06d3152ebd/include/bitcoin/bitcoin/chain/block.hpp#L167) exists. Do you prefer a separate function then?

Reply:
Parameterizing hash() creates a lot of potential issues.